TsJournalTrello.gs

```diff
@@ -0,0 +1,361 @@
+  var orgFound = false
+    
+  //Loop through the rows of data
+  orgFound = data.some(function(row)  {
+      
+    //check if the company name matches the orgName
+    var orgNameSearch = row[ORG_NAME_COL_]
+      
+    if (orgNameSearch === orgName) {
+    
+      //Get the timesheet and Trello Board Url
+      timesheetUrl = row[ORG_TS_COL_]
+      trelloBoardUrl = row[TRELLO_COL_]
+      return true
+      
+      }
+      
+  })   
+    
+  if (!orgFound) {
+    
+    Log_.warning('Organisation Name: ' + orgName + ' not found in Org Spreadsheet')
+    return
+      
+  } else {
+    
+    Log_.info('Organisation Name: ' + orgName + ' found in Org Spreadsheet')
+      
+  }
+   
+    
+  //Check the Trello Board Exists  
+  if (trelloBoardUrl === '') {
+    
+    Log_.warning('No Trello Board URL found in Org Spreadsheet')
+    return
+      
+  } else {
+      
+    //Get the Trello Board Id from the wedsite json return  
+    trelloBoardId = getTrelloBoardId_(trelloBoardUrl)
+    
+    if (trelloBoardId == null) {
+      Log_.warning('No new trello board Id found for ' + trelloBoardUrl)
+      return
+    }    
+    Log_.info('Trello Card Id: ' + trelloBoardId)
+        
+  } //if (trelloBoardUrl === '')
+      
+  //Open the Journal, loop through the paragraphs until the Heading 1 is found
+  //The next paragraph is the Trello Card Name, check it hasnt already been processed.
+  //If it hasnt, change the style to Heading 2 and add a Hyperlink to the Trello Card 
+  var journal = DocumentApp.openById(DocId)
+  var DocUrl = journal.getUrl()
+  var npBookmark = null
+ 
+  if (journal === null) { 
+  
+    Log_.warning('Invalid Journal ID' + DocId)
+   
+    return
+  } else {
+   
+    //Process the journal, find the next tello card task, hyperlink the heading
+    var [trelloCardTitle, npBookmarkId] = processJournal_(journal, trelloBoardId)
+          
+    if (trelloCardTitle == null) {
+      Log_.warning('Either no trello card title found or trello card found on trello board')
+      return
+    } else if (npBookmarkId == null) {
+      Log_.warning('No Bookmark Found')    
+      return
+    } else {
+      //Add the link to the timesheet
+      processTimesheet_(timesheetUrl, npBookmarkId, DocUrl, trelloCardTitle)    
+    }
+  }
+                
+  return SUCCESS  
+}// linkHeaders() 
+
+
+function processJournal_(journal, trelloBoardId) {
+          
+  var paragraphs = journal.getBody().getParagraphs()
+  var ProcessJournalReturn = null
+  var Heading1Paragraphs = []
+  
+  for(var i in paragraphs) { 
+    if (paragraphs[i].getHeading() === DocumentApp.ParagraphHeading.HEADING1) {
+      Heading1Paragraphs.push(i)
+    }
+  }
+  
+  var lastHeading1Paragraph = Heading1Paragraphs[Heading1Paragraphs.length -1]
+  var trelloCardParagraph = paragraphs[Number(lastHeading1Paragraph) + 1]
+  var trelloCardTitle = trelloCardParagraph.getText()
+  
+  if (trelloCardParagraph.getHeading() !== DocumentApp.ParagraphHeading.HEADING2) {
```

Comment:
[The paragraph shouldn't have to be heading 2 yet](https://docs.google.com/document/d/12lKDk6A-P2-JGo4vhz1fE_33vC_MvxC1hbcUpfV5R2o/edit#bookmark=id.i93cngih95kf), although saying that I didn't have to set it to H2 but it worked??

Reply:
This is checking that the paragraph hasn't already been processed in a previous run, if it hasnt been processed i.e. changed to heading 2 and had the urllink added, then process.